libraries/AP_Scheduler/AP_Scheduler.cpp

```diff
@@ -89,96 +90,87 @@ void AP_Scheduler::tick(void)
-                    hal.util->perf_begin(_perf_counters[i]);
-                }
-                _tasks[i].function();
-                if (_debug > 1 && _perf_counters && _perf_counters[i]) {
-                    hal.util->perf_end(_perf_counters[i]);
-                }
-                current_task = -1;
-
-                // record the tick counter when we ran. This drives
-                // when we next run the event
-                _last_run[i] = _tick_counter;
-
-                // work out how long the event actually took
-                now = AP_HAL::micros();
-                uint32_t time_taken = now - _task_time_started;
-
-                if (time_taken > _task_time_allowed) {
-                    // the event overran!
-                    if (_debug > 4) {
-                        ::printf("Scheduler overrun task[%u-%s] (%u/%u)\n",
-                                 (unsigned)i,
-                                 _tasks[i].name,
-                                 (unsigned)time_taken,
-                                 (unsigned)_task_time_allowed);
-                    }
-                }
-                if (time_taken >= time_available) {
-                    goto update_spare_ticks;
-                }
-                time_available -= time_taken;
+        // run it
+        if (_task_time_allowed <= time_available) {
+            _task_time_started = task_time_buf;
+            current_task = i;
+            const bool bPerfCntrs = _debug > 1 && _perf_counters && _perf_counters[i];
+            if (bPerfCntrs) {
+                hal.util->perf_begin(_perf_counters[i]);
+            }
+            _tasks[i].function();
+            if (bPerfCntrs) {
+                hal.util->perf_end(_perf_counters[i]);
+            }
+            current_task = -1;
+
+            // record the tick counter when we ran. This drives
+            // when we next run the event
+            _last_run[i] = _tick_counter;
+
+            // work out how long the event actually took
+            task_time_buf = AP_HAL::micros();
+            const uint32_t time_taken = task_time_buf - _task_time_started;
+
+            // the event overran!
+            if (_debug > 4 && time_taken > _task_time_allowed) {
+                ::printf("Scheduler overrun task[%u-%s] (%u/%u)\n",
+                            (unsigned)i,
+                            _tasks[i].name,
+                            (unsigned)time_taken,
+                            (unsigned)_task_time_allowed);
             }
+            
+            if (time_taken >= time_available) {
+                time_available = 0;
+                break;
+            }
+            time_available -= time_taken;
         }
     }
 
     // update number of spare microseconds
     _spare_micros += time_available;
-
-update_spare_ticks:
     _spare_ticks++;
-    if (_spare_ticks == 32) {
-        _spare_ticks /= 2;
-        _spare_micros /= 2;
-    }
+    
```

Comment:
I do not get this change...
I do not think that it is more readable.
I do not think that it runs faster.

What is the point here ?

Reply:
just shorter, but if you don't like

Reply:
Please revert it please